go/common/crypto/slip10/slip0010.go

```diff
@@ -0,0 +1,93 @@
+// Package slip10 implements the SLIP-0010 private key derivation
+// scheme for Ed25519.
+package slip10
+
+import (
+	"crypto/hmac"
+	"crypto/sha512"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature/signers/memory"
+)
+
+const (
+	// SeedSize is the BIP-0039 seed byte sequence size in bytes.
+	SeedSize = 64
+
+	// ChainCodeSize is the size of a SLIP-0010 chain code in bytes.
+	ChainCodeSize = 32
+)
+
+var curveConstant = []byte("ed25519 seed")
+
+// ChainCode is a SLIP-0010 chain code.
+type ChainCode [ChainCodeSize]byte
+
+// NewMasterKey derives a master key and chain code from a seed byte sequence.
+func NewMasterKey(seed []byte) (signature.Signer, ChainCode, error) {
+	// 1. Generate a seed byte sequence S of 512 bits according to BIP-0039.
+	if len(seed) != SeedSize {
+		return nil, ChainCode{}, fmt.Errorf("slip0010: invalid seed")
```

Comment:
Could you replace `slip0010` in this and the following 3 error messages with `slip10`?

Reply:
I should have just left the package name as is.